resources/experimental-feature.md

```diff
@@ -0,0 +1,89 @@
+
+⚠️ To ensure the users understand the feature is experimental, the flag needs to have the `experimental` prefix.
+
+## Rules and expectations
+
+- The API and the behavior of an experimental feature can break between two minor versions of Meilisearch.
+- The whole experimental feature process described in this guide can break between 2 minor versions of Meilisearch.
+- Providing a feature as “experimental” does not guarantee it will be stable one day: the newly introduced experimental features or improvements can be removed in the future.
+- The experimental features are supposed to be unstable regarding the usage but **not buggy**.
+The features will be released like any stable feature with a whole test suite and serious code reviews, something we don’t have the time to do when providing a prototype, for example.
+However, the experimental feature process can be an opportunity to test an improvement (for example, regarding performance) we are not comfortable putting immediately into production. In this case, more bugs and/or API changes are expected.
+
+## Communication with users
+
+For each experimental feature:
+- In the [product repository](https://github.com/meilisearch/product/discussions)
+  - A dedicated GitHub discussion is open
+  - The discussion should be clear about the following information
+      - Our motivations: why this feature is unstable?
+      - How to use the feature: how to activate it? Do we need to do a migration with a dump?
+      - Our future plan: what do we expect to make it stable? When will the feature potentially be stable?
+  - The discussion should not stay in a stale state. Users need to understand they can interact in the discussion and get quick answers. If needed, the discussion will be updated regularly.
+- Using the Meilisearch CLI: the `--help` command in the Meilisearch binary will redirect to the related GitHub discussion and warn about the unstable state of the features.
+- In the documentation: a small dedicated page should be done about the purpose of the experimental feature and redirect to the related GitHub discussion.
+
+## Some warnings about the usage
+
+- Experimental features are not recommended to be used in production.
+- For some provided experimental features, re-indexing will be needed. For this, more than only restarting the search engine, using a dump will be needed to enforce re-indexing.
+This means, in some situations, using a dump might be needed to activate an experimental feature but also to deactivate it. The users will be clearly informed in the related discussion how to activate the feature
+
+> ⚠️ When deactivating this kind of experimental feature, since our process is not mature yet, users can have some issues with their DB, even when using a dump.<br>
+> We recommend users always save their data (with snapshots and/or dumps) before starting to activate experimental features. Of course, the safest method is to use a script to feed Meilisearch.
```

Comment:
```suggestion
## Usage warnings

- Experimental features are not recommended for production environments
- Some experimental features, might required re-indexing. In these cases, users will have to use a dump to activate and deactivate an experimental feature. Users will be clearly informed about this in the related GitHub discussion

> ⚠️ Since this process is not mature yet, users might experience issues with their DB when deactivating these features even when using a dump.<br>
> We recommend users always save their data (with snapshots and/or dumps) before activating experimental features.
```